django/utils/functional.py

```diff
@@ -288,15 +288,25 @@ def __init__(self):
         self._wrapped = empty
 
     def __getattribute__(self, name):
-        value = super().__getattribute__(name)
-        # If attribute is a proxy method, raise an AttributeError to call
-        # __getattr__() and use the wrapped object method.
-        if not getattr(value, "_mask_wrapped", True):
-            raise AttributeError
```

Comment:
Hmm might I also suggest this instead?

```python
def __getattribute__(self, name):
    if name == '_wrapped':
            return super().__getattribute__(name)
    value = super().__getattribute__(name)
    # If attribute is a proxy method, raise an AttributeError to call
    # __getattr__() and use the wrapped object method.
    if not getattr(value, "_mask_wrapped", True):
         raise AttributeError
    return value

__getattr__ = new_method_proxy(getattr)
```

Basically from what I understand, this bug happens on cases where the wrapped objects`getattr` returns a `False` value, not necessarily a value. For example if you returned [1] instead of an empty list, the test would pass. This misses the `AttributeError` bellow. 

In both cases `if not found or not getattr(value, "_mask_wrapped", True):` and `if not getattr(value, "_mask_wrapped", True):` will return `[]` and go into the condition. Both implementations basically safeguard `_wrapped` to be returned and not go into the loop

Reply:
Interesting. This looks much simpler and seems to work well.

Reply:
Of course :facepalm: @2019342a Thanks :+1: